src/main/java/org/osc/manager/rest/server/api/DeviceApis.java

```diff
@@ -0,0 +1,391 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.manager.rest.server.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.log4j.Logger;
+import org.osc.manager.ism.api.IsmDeviceApi;
+import org.osc.manager.ism.entities.DeviceEntity;
+import org.osc.manager.ism.entities.DeviceMemberEntity;
+import org.osc.manager.rest.server.SecurityManagerServerRestConstants;
+import org.osc.sdk.manager.element.ApplianceSoftwareVersionElement;
+import org.osc.sdk.manager.element.DistributedApplianceElement;
+import org.osc.sdk.manager.element.DomainElement;
+import org.osc.sdk.manager.element.ManagerDeviceElement;
+import org.osc.sdk.manager.element.VirtualSystemElement;
+import org.osc.sdk.manager.element.VirtualizationConnectorElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.transaction.control.TransactionControl;
+
+@Component(service = DeviceApis.class)
+@Path(SecurityManagerServerRestConstants.SERVER_API_PATH_PREFIX + "/devices")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class DeviceApis {
+
+    private static final Logger logger = Logger.getLogger(DeviceApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+    private IsmDeviceApi api;
+
+    public static final class VSElement implements VirtualSystemElement {
```

Comment:
I dont think you need this class. See my previous comment about how to reuse the code without needing this.

Reply:
OK